src/Kdyby/Translation/Dumper/DatabaseDumper.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Kdyby\Translation\Dumper;
+
+use Kdyby\Translation\Helpers;
+use Symfony\Component\Translation\Dumper\DumperInterface;
+
+abstract class DatabaseDumper implements DumperInterface
+{
+
+    /** @var string */
+    protected $table = 'translations';
+
+    /** @var string */
+    protected $key = 'key';
+
+    /** @var string */
+    protected $locale = 'locale';
+
+    /** @var string */
+    protected $message = 'message';
+
+    /** @var string */
+    protected $updatedAt = 'updated_at';
+
+    /**
+     * @param string $table
+     */
+    public function setTableName($table)
+    {
+        $this->table = $table;
+    }
+
+    /**
+     * @param string $key
+     * @param string $locale
+     * @param string $message
+     * @param string $updatedAt
+     */
+    public function setColumns($key, $locale, $message, $updatedAt)
+    {
+        $this->key = $key;
+        $this->locale = $locale;
+        $this->message = $message;
+        $this->updatedAt = $updatedAt;
+    }
+
+    /**
+     * Dumps the message catalogue.
+     *
+     * @param \Symfony\Component\Translation\MessageCatalogue $messages The message catalogue
+     * @param array $options Options that are used by the dumper
+     */
+    public function dump(\Symfony\Component\Translation\MessageCatalogue $messages, $options = array())
+    {
+        $messagesArray = $messages->all();
+        Helpers::flatten($messagesArray);
+        $locale = $messages->getLocale();
+        $keys = array_keys($messagesArray);
+
+        $this->beginTransaction();
```

Comment:
on to není jeden dotaz. Protože se tam v cyklu provádí hodně updatů a insertů. A když chci, aby se všechny provedly rychle, uzavřu je do transakce.